library/auth/src/main/java/com/epmedu/animeal/auth/AuthAPI.kt

```diff
@@ -0,0 +1,92 @@
+package com.epmedu.animeal.auth
+
+import com.amplifyframework.auth.AuthUserAttribute
+import com.amplifyframework.auth.AuthUserAttributeKey
+import com.amplifyframework.auth.cognito.options.AWSCognitoAuthSignInOptions
+import com.amplifyframework.auth.cognito.options.AuthFlowType
+import com.amplifyframework.auth.options.AuthSignUpOptions
+import com.amplifyframework.core.Amplify
+
+class AuthAPI {
+    fun signUp(
+        phone: String,
+        password: String,
+        handler: AuthRequestHandler
+    ) {
+        val attrs = mapOf(
+            AuthUserAttributeKey.phoneNumber() to phone,
+        )
+
+        val options = AuthSignUpOptions.builder()
+            .userAttributes(attrs.map { AuthUserAttribute(it.key, it.value) })
+            .build()
+
+        Amplify.Auth.signUp(
+            phone,
+            password,
+            options,
+            {
+                handler.onSuccess()
```

Comment:
You don't need to create lambdas. Just use function references like this: `handler::onSuccess,
            handler::onError`